core/src/main/java/io/questdb/griffin/engine/functions/groupby/ApproxPercentileDoubleGroupByFunction.java

```diff
@@ -0,0 +1,198 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.ArrayColumnTypes;
+import io.questdb.cairo.ColumnType;
+import io.questdb.cairo.map.MapValue;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.SymbolTableSource;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.*;
+import io.questdb.std.Numbers;
+import io.questdb.std.ObjList;
+import io.questdb.std.histogram.org.HdrHistogram.DoubleHistogram;
+
+public class ApproxPercentileDoubleGroupByFunction extends DoubleFunction implements GroupByFunction, TernaryFunction {
+    private final int funcPosition;
+    private final Function exprFunc;
+    private final Function percentileFunc;
+    // specifies the precision for the recorded values (between 0 and 5, defaults to 3).
+    // trade-off between memory usage and accuracy.
+    private final Function precisionFunc;
+    private final ObjList<DoubleHistogram> histograms = new ObjList<>();
+    private int histogramIndex;
+    private int valueIndex;
+
+    Function defaultPrecisionFunc = new IntFunction() {
+        final int defaultPrecision = 3;
+
+        @Override
+        public int getInt(Record rec) {
+            return defaultPrecision;
+        }
+
+        @Override
+        public boolean isConstant() {
+            return true;
+        }
+
+        @Override
+        public void toPlan(PlanSink sink) {
+            sink.val(defaultPrecision);
+        }
+    };
+
+    public ApproxPercentileDoubleGroupByFunction(int funcPosition, Function exprFunc, Function percentileFunc) {
+        this.funcPosition = funcPosition;
+        this.exprFunc = exprFunc;
+        this.percentileFunc = percentileFunc;
+        this.precisionFunc = defaultPrecisionFunc;
+    }
+
+    public ApproxPercentileDoubleGroupByFunction(int funcPosition, Function exprFunc, Function percentileFunc, Function precisionFunc) {
+        this.funcPosition = funcPosition;
+        this.exprFunc = exprFunc;
+        this.percentileFunc = percentileFunc;
+        this.precisionFunc = precisionFunc;
+    }
+
+    @Override
+    public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+        exprFunc.init(symbolTableSource, executionContext);
```

Comment:
nit: these three lines of code should be replaced with `TernaryFunction.super.init(symbolTableSource, executionContext);`